lib/ProductOpener/Display.pm

```diff
@@ -7870,6 +7870,12 @@ CSS
 
 	$request_ref->{canon_url} = product_url($product_ref);
 
+	if ($lc eq 'en') {
+		my $subdomain = 'https://world.openfoodfacts.org';
+		$request_ref->{canon_url} = $subdomain;
+		$request_ref->{canon_url} .= product_url($product_ref);
```

Comment:
That's not going to work, product_url() gives back the full URL.
You can try it with docker, it will give you a broken url in the <link rel="canonical"> tag.

product_url() gives you an url like https://uk.(openfoodfacts.org|productopener.localhost)/... 

and instead we want the same url, with uk (or us etc.) replaced by world, and the rest of the url as it was.

e.g. it can be done with a regex to replace https://(something without a dot). with https://world.

$request_ref->{canon_url} =~ s/https:\/\/([^\.])\./https:\/\/world./ ;